test/jdk/com/sun/net/httpserver/simpleserver/FileServerHandlerTest.java

```diff
@@ -82,6 +84,106 @@ public void testNotImplementedRequestMethod(String requestMethod) throws Excepti
         assertEquals(exchange.rCode, 501);
     }
 
+    @DataProvider
+    public Object[][] validRangeHeaders() {
+        return new Object[][]{
+                // start-end
+                {1000L, "bytes=0-499", new FileServerHandler.RangeEntry(0, 499)},
+                {1000L, "bytes=500-999", new FileServerHandler.RangeEntry(500, 999)},
+                {500L,  "bytes=100-199", new FileServerHandler.RangeEntry(100, 199)},
+
+                // start-
+                {1000L, "bytes=0-", new FileServerHandler.RangeEntry(0, 999)},
+                {1000L, "bytes=500-", new FileServerHandler.RangeEntry(500, 999)},
+                {500L,  "bytes=250-", new FileServerHandler.RangeEntry(250, 499)},
+
+                // -length
+                {1000L, "bytes=-500", new FileServerHandler.RangeEntry(500, 999)},
+                {1000L, "bytes=-1", new FileServerHandler.RangeEntry(999, 999)},
+                {500L,  "bytes=-50", new FileServerHandler.RangeEntry(450, 499)},
+
+                // single byte ranges
+                {1000L, "bytes=0-0", new FileServerHandler.RangeEntry(0, 0)},
+                {1000L, "bytes=999-999", new FileServerHandler.RangeEntry(999, 999)},
+                {500L,  "bytes=499-499", new FileServerHandler.RangeEntry(499, 499)},
+
+                // multiple ranges
+                {
+                        1000L, "bytes=0-400,500-999",
+                        new FileServerHandler.RangeEntry(0, 400),
+                        new FileServerHandler.RangeEntry(500, 999)
+                },
+                {
+                        1000L, "bytes=0-,-200",
+                        new FileServerHandler.RangeEntry(0, 999),
+                        new FileServerHandler.RangeEntry(800, 999)
+                },
+                {
+                        500L, "bytes=0-99,200-",
+                        new FileServerHandler.RangeEntry(0, 99),
+                        new FileServerHandler.RangeEntry(200, 499)
+                },
+                // merging overlapping or contiguous ranges
+                {
+                        1000L, "bytes=0-499,400-599,600-999",
+                        new FileServerHandler.RangeEntry(0, 999)
+                },
+                {
+                        1000L, "bytes=0-499,500-699,550-999",
+                        new FileServerHandler.RangeEntry(0, 999)
+                },
+                {
+                        1000L, "bytes=0-499,500-999",
+                        new FileServerHandler.RangeEntry(0,999)
+                },
+                {
+                        1000L, "bytes=100-199,50-149",
+                        new FileServerHandler.RangeEntry(50,199)
+                },
+                {
+                        1000L, "bytes=100-199,300-399,150-350",
+                        new FileServerHandler.RangeEntry(100,399)
+                },
```

Comment:
Can we add a test with duplicated ranges too? e.g.

`0-99,200-399,0-99,400-499,200-499`

Reply:
I added a test case for duplicated ranges, thanks.